### PR TITLE
chore: bump CAPI to v1.5.0

### PR DESCRIPTION
**Pull request by richardcase**
**What type of PR is this?**

/kind other

**What this PR does / why we need it**:

This bumps main to use v1.5.0-rc1 of CAPI. It also bumps to use Go v1.20 and controller-runtime v0.15.0.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Bump CAPI to v1.5.0 (and associated packages)
```

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
     name: lint
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - uses: actions/setup-go@v3
+      - uses: actions/setup-go@v4
         with:
-          go-version: 1.19
-          check-latest: true
+          go-version: 1.20.6
```
Comment (cpanato) [inline .github/workflows/lint.yml:15]:
```suggestion
          go-version: '1.20'
```

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
   golangci:
     name: lint
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
-      - uses: actions/setup-go@v3
+      - uses: actions/setup-go@v4
         with:
-          go-version: 1.19
-          check-latest: true
```
Comment (cpanato) [inline .github/workflows/lint.yml:13]: keep this

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -1,17 +1,20 @@
...
+          go-version: 1.20.6
+          cache: false
       - name: golangci-lint
-        uses: golangci/golangci-lint-action@v3.2.0
+        uses: golangci/golangci-lint-action@v3
         with:
-          version: v1.50.1
+          version: v1.53.3
```
Comment (cpanato) [inline .github/workflows/lint.yml:20]:
```suggestion
          version: v1.53
```

then we dont need to update for patch releases

File: Dockerfile [hunk 4]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 
 # Build the manager binary
-FROM golang:1.19.10@sha256:83f9f840072d05ad4d90ce4ac7cb2427632d6b89d5ffc558f18f9577ec8188c0 as builder
+FROM public.ecr.aws/docker/library/golang:1.20.6 as builder
```
Comment (cpanato) [inline Dockerfile:16]: what is the advantage of using that? also can we add the digest as well?

**Review by cpanato — CHANGES_REQUESTED** [review]
looks very cool, thanks for doing that

just some minor comments and requests

File: .github/workflows/lint.yml (hunk 3 above)
Comment (richardcase) [reply .github/workflows/lint.yml]: Changed

**Review by cpanato — APPROVED** [review]
yayy
/lgtm

**Pull request merged**